frontend/src/components/editor/chrome/panels/selection-panel.tsx

```diff
@@ -0,0 +1,15 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { SlotNames } from "@/core/slots/slots";
+import { Fill, Slot } from "@marimo-team/react-slotz";
+import type { PropsWithChildren } from "react";
+
+export const DataSelectionPanelSlot: React.FC = () => {
+  return <Slot name={SlotNames.DATA_SELECTION} />;
```

Comment:
if we want to render the pinning and sidebar logic here, instead can do `useSlot()`. and check if the list is empty to just render `null`